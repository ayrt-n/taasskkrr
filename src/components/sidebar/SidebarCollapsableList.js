import React, { useState } from 'react';
import chevron from '../../assets/icons/chevron-right.svg';

function SidebarCollapsableList({ title, children }) {
  const [collapsed, setCollapsed] = useState(false);
  const collapseList = () => { setCollapsed(!collapsed) }

  return(
    <div
      className={`Sidebar-collapsable-list ${collapsed ? "collapsed" : ""}`}
      data-testid="collapsable-list"
    >
      <div className="SidebarItem" onClick={collapseList} data-testid="collapse-list-div">
        <img
          src={chevron}
          className="SidebarItem-icon"
          alt=""
          style={collapsed ? {} : {transform: "rotate(90deg)"}} />
        <div className="SidebarItem-title">{title}</div>
      </div>
      <div className={"Sidebar-collapsable-items"}>
        {children}
      </div>
    </div>
  );
}

export default SidebarCollapsableList;
