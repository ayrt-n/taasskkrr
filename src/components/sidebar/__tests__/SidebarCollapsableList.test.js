import React from 'react';
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event';
import SidebarCollapsableList from '../SidebarCollapsableList';

// Set up props
const title = 'Collapsable List'

describe('SidebarCollapsableList component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <SidebarCollapsableList title={title}>
        <div>Test 1</div>
        <div>Test 2</div>
        <div>Test 3</div>
      </SidebarCollapsableList>
    );

    expect(container).toMatchSnapshot();
  });

  describe('on click', () => {
    it('adds collapsed class to collapsable list', () => {
      render(<SidebarCollapsableList title={title} />);
      const collapsableList = screen.getByTestId('collapsable-list');
      const collapsableListDiv = screen.getByTestId('collapse-list-div');

      expect(collapsableList.className).not.toContain('collapsed');
      userEvent.click(collapsableListDiv);
      expect(collapsableList.className).toContain('collapsed');
    });
  })
});