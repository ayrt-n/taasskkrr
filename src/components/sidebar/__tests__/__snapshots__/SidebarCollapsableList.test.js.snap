// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarCollapsableList component renders correctly 1`] = `
<div>
  <div
    class="Sidebar-collapsable-list "
    data-testid="collapsable-list"
  >
    <div
      class="SidebarItem"
      data-testid="collapse-list-div"
    >
      <img
        alt=""
        class="SidebarItem-icon"
        src="test-file-stub"
        style="transform: rotate(90deg);"
      />
      <div
        class="SidebarItem-title"
      >
        Collapsable List
      </div>
    </div>
    <div
      class="Sidebar-collapsable-items"
    >
      <div>
        Test 1
      </div>
      <div>
        Test 2
      </div>
      <div>
        Test 3
      </div>
    </div>
  </div>
</div>
`;
